projects/experimental/directives/header/header.directive.ts

```diff
@@ -0,0 +1,33 @@
+import {Directive, Inject, Input} from '@angular/core';
+import {TuiDirectiveStylesService} from '@taiga-ui/cdk';
+import {TuiSizeXXL, TuiSizeXXS} from '@taiga-ui/core';
+import {
+    tuiAvatarOptionsProvider,
+    tuiButtonOptionsProvider,
+} from '@taiga-ui/experimental/components';
+import {tuiBadgeOptionsProvider} from '@taiga-ui/kit';
+
+import {TuiHeaderComponent} from './header.component';
+
+@Directive({
+    selector: '[tuiHeader]',
+    providers: [
+        tuiAvatarOptionsProvider({size: 's'}),
+        tuiButtonOptionsProvider({size: 's'}),
+        tuiBadgeOptionsProvider({size: 'm'}),
+    ],
+    host: {
+        tuiHeader: '',
+        '[attr.data-size]': 'size || "s"',
+    },
+})
+export class TuiHeaderDirective {
+    @Input('tuiHeader')
+    size: TuiSizeXXL | TuiSizeXXS | '' = 's';
+
+    constructor(
+        @Inject(TuiDirectiveStylesService) directiveStyles: TuiDirectiveStylesService,
+    ) {
+        directiveStyles.addComponent(TuiHeaderComponent);
```

Comment:
Let's use the new approach from now on:
https://github.com/taiga-family/taiga-ui/blob/main/projects/experimental/components/textfield/label.directive.ts#L13